From oliver@neukum.org  Wed Nov 12 11:17:26 2008
From: Oliver Neukum <oliver@neukum.org>
Date: Fri, 1 Feb 2008 22:44:40 +0100
Subject: USB: power availability check for berry_charge
To: Greg KH <greg@kroah.com>, linux-usb@vger.kernel.org
Message-ID: <200802012244.41310.oliver@neukum.org>
Content-Disposition: inline


this introduces a sanity check into berry_charge to give up before
damage is done if we lack juice.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/berry_charge.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/usb/misc/berry_charge.c
+++ b/drivers/usb/misc/berry_charge.c
@@ -123,6 +123,11 @@ static int berry_probe(struct usb_interf
 {
 	struct usb_device *udev = interface_to_usbdev(intf);
 
+	if (udev->bus_mA < 500) {
+		dbg(&udev->dev, "Not enough power to charge available\n");
+		return -ENODEV;
+	}
+
 	dbg(&udev->dev, "Power is set to %dmA\n",
 	    udev->actconfig->desc.bMaxPower * 2);
 
